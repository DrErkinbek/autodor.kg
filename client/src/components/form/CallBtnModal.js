import React, { useState } from 'react';
import emailjs from 'emailjs-com';
import M from 'materialize-css/dist/js/materialize.min.js';

const CallBtnModal = () => {
    const [customer, setCustomer ] = useState({
        customer_name: '',
        customer_number: ''
    });
    const { customer_name, customer_number } = customer;

    const resetForm = () => setCustomer({
        customer_name: '',
        customer_number: ''
    });

    const onChange = e => setCustomer({ ...customer, [e.target.name]: e.target.value });

    const onSubmit = e => {
        e.preventDefault();

        let CallBtnParams = {
            customer_name: customer_name,
            customer_number: customer_number
        }
        emailjs.send(
            'service_gn5xg5p',
            'template_bqtju7m',
            CallBtnParams,
            'user_LluexlTTpyxIb1Jzgoe6a'
        )
        resetForm();
        M.toast({ html: 'Спасибо, мы вам обратно звоним.', classes: 'rounded green lighten-1'});
    }

    return(
        <div id="call-modal" className="modal" style={modalStyle}>
            <div className="modal-content">
                    <div className="modal-header right">
                        <a href="#!" className="modal-close">
                            <i className="small material-icons">close</i>
                        </a>
                    </div>
                    <div className="col s12 m6 l8">
                        <form onSubmit={onSubmit}>
                            <div className="center">
                                <p>Закажите обратный звонок</p>
                            </div>
                            <div className="input-field col s6">
                                <input id="icon_prefix" type="text" className="validate"
                                onChange={onChange} minLength="4"
                                placeholder="Ваша имя" name="customer_name" value={customer_name}
                                required />
                            </div>
                            <div className="input-field col s6">
                                <input id="icon_telephone" type="tel" className="validate"
                                onChange={onChange} minLength="9"
                                placeholder="Телефон номер" name="customer_number" value={customer_number}
                                required />
                            </div>
                            <button className="blue btn-small right">Отправить</button>
                        </form>
                    </div>
            </div>
        </div>
    )
}

const modalStyle = {
    width: "75%",
    height: "75%"
}

export default CallBtnModal;