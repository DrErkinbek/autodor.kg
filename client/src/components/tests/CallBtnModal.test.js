import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import CallBtnModal from '../form/CallBtnModal';

it('CallBtnModal renders without crashing ', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CallBtnModal />, div);
});

it('CallBtnModal renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<CallBtnModal />).toJSON();
  
    expect(tree).toMatchSnapshot();
});