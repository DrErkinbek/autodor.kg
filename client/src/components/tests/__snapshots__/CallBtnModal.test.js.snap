// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallBtnModal renders correctly and equal to snapshot 1`] = `
<div
  className="modal"
  id="call-modal"
  style={
    Object {
      "height": "75%",
      "width": "75%",
    }
  }
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header right"
    >
      <a
        className="modal-close waves-effect"
        href="#!"
      >
        <i
          className="small material-icons"
        >
          close
        </i>
      </a>
    </div>
    <div
      className="col s12 m6 l8"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="center"
        >
          <p>
            Закажите обратный звонок
          </p>
        </div>
        <div
          className="input-field col s6"
        >
          <input
            className="validate"
            id="icon_prefix"
            minLength="4"
            name="customer_name"
            onChange={[Function]}
            placeholder="Ваша имя"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="input-field col s6"
        >
          <input
            className="validate"
            id="icon_telephone"
            minLength="9"
            name="customer_number"
            onChange={[Function]}
            placeholder="Телефон номер"
            required={true}
            type="tel"
            value=""
          />
        </div>
        <button
          className="blue waves-light btn-small right"
        >
          Отправить
        </button>
      </form>
    </div>
  </div>
</div>
`;
